CodeEdit/Features/NavigatorSidebar/ProjectNavigator/OutlineView/ProjectNavigatorViewController.swift

```diff
@@ -86,6 +86,10 @@ final class ProjectNavigatorViewController: NSViewController {
         fatalError()
     }
 
+    @objc func revealFile(_ sender: Any) {
```

Comment:
Everything works great except this has one small bug. When `Settings.shared.preferences.general.revealFileOnFocusChange` is set to `false` the file isn't revealed. This is because `select(by id: TabBarItemID, from collection: [CEWorkspaceFile])` won't reveal the file if that setting is false.

To fix this is really simple. Add a parameter to `select` called `forceReveal` that defaults to false. It'd look like `private func select(by id: TabBarItemID, from collection: [CEWorkspaceFile], forceReveal: Bool = false)`. That forceReveal parameter should override the setting check on line `330`. Then call `select` from this function instead of `updateSelection`.

Then just document this function and I'd be happy to accept this change!

Reply:
Thank you for pointing me out. I updated it to forcefully reveal the file if the setting is turned off.